### PR TITLE
Fix the labeling of subdomain boundary faces

### PR DESCRIPTION
**Pull request by Laar**
The current check is to look if the same (by id) global nodes are both
side of the face. This identical to checking their coordinates but
independent of the dimension. This differs from the previous version
where no only the node coordinates were checked, but also their
ordering (which does not need to be the same on both sides).

### CODE REVIEW

File: kernel/Base/Face.cpp [hunk 1]
```diff
@@ -89,71 +89,27 @@ Face::Face(Element* ptrElemL, const std::size_t& localFaceNumberL,
...
+                localFaceNumberL);
+        std::vector<std::size_t> globalRightNodes =
+            ptrElemR->getPhysicalGeometry()->getGlobalFaceNodeIndices(
+                localFaceNumberR);
+        std::sort(globalLeftNodes.begin(), globalLeftNodes.end());
+        std::sort(globalRightNodes.begin(), globalRightNodes.end());
+        if (globalLeftNodes != globalRightNodes) {
+            setFaceType(Geometry::FaceType::PERIODIC_BC);
+        }
+    }
```
Comment (JensWehner) [inline kernel/Base/Face.cpp:107]: this should be a function.

**Review by JensWehner — CHANGES_REQUESTED** [review]
Nice change, makes the code more readable, but it should still be a function.

**Review by JensWehner — CHANGES_REQUESTED** [review]
Nice, changes, make the code more readable. Can you refactor it into a function

**Comment by Laar** [comment]
Part of the reason for leaving it inline was that the function does not determine what type of periodic boundary face it is, i.e., subdomain or not. Currently I'm not wanting to change too much to prevent it accidentally breaking.

I hope that if we refactor the Mesh that this will be improved a lot.

File: kernel/Base/Face.cpp [hunk 2]
```diff
@@ -364,71 +308,15 @@ void Face::addElement(Element* ptrElementR, std::size_t localFaceNumberR) {
     std::vector<std::size_t> localRightNodes =
         ptrElementR->getPhysicalGeometry()->getLocalFaceNodeIndices(
             localFaceNumberR);
+
+    if (detirmineIsPeriodicBoundaryFace()) {
```
Comment (JensWehner) [inline kernel/Base/Face.cpp:312]: fix spelling pls.
```suggestion
    if (determineIsPeriodicBoundaryFace()) {
```

File: kernel/Base/Face.cpp [hunk 3]
```diff
@@ -89,71 +89,15 @@ Face::Face(Element* ptrElemL, const std::size_t& localFaceNumberL,
     std::vector<std::size_t> localRightNodes =
         ptrElemR->getPhysicalGeometry()->getLocalFaceNodeIndices(
             localFaceNumberR);
+
+    if (detirmineIsPeriodicBoundaryFace()) {
```
Comment (JensWehner) [inline kernel/Base/Face.cpp:93]:
```suggestion
    if (determineIsPeriodicBoundaryFace()) {
```

File: kernel/Base/Face.cpp [hunk 4]
```diff
@@ -455,6 +343,21 @@ const std::vector<Base::Node*> Face::getNodesList() const {
 
     return ptrNodesAtFace;
 }
+
+bool Face::detirmineIsPeriodicBoundaryFace() {
```
Comment (JensWehner) [inline kernel/Base/Face.cpp:347]:
```suggestion
bool Face::determineIsPeriodicBoundaryFace() {
```

File: kernel/Base/Face.h [hunk 5]
```diff
@@ -377,6 +377,10 @@ class Face final : public Geometry::FaceGeometry, public FaceData {
 
     std::vector<std::size_t> numberOfConformingDOFOnTheFace_;
     std::size_t faceID_;
+
+    /// Determine whether this face is a face on the periodic boundary. Is
+    /// independent of the faceType.
+    bool detirmineIsPeriodicBoundaryFace();
```
Comment (JensWehner) [inline kernel/Base/Face.h:383]: should be const 
```suggestion
    bool determineIsPeriodicBoundaryFace()const;
```

**Review by JensWehner — CHANGES_REQUESTED** [review]
@Laar good refactor, just a small spelling mistake and `const` stuff

**Comment by Laar** [comment]
> @Laar good refactor, just a small spelling mistake and `const` stuff

@JensWehner Done, needed a little more formatting (and the `const` in the cpp file).

**Review by JensWehner — APPROVED** [review]
LGTM

**Pull request merged**